forge/experiment_tools.py

```diff
@@ -95,19 +95,35 @@ def load_from_checkpoint(checkpoint_dir, checkpoint_iter, path_prefix='',
     if model_kwargs is not None:
         data.update(model_kwargs)
 
-    model = load(path_prefix + F.model_config, F, **data)
-    all_train_vars_after = set(tf.trainable_variables())
-    model_vars = list(all_train_vars_after - all_train_vars_before)
-
     checkpoint_path = osp.join(checkpoint_dir, 'model.ckpt-{}'.format(checkpoint_iter))
+    model_config = path_prefix + F.model_config
+
+    if mode is None:
+        checkpoint = checkpoint_path
+        model = load(model_config, F)
```

Comment:
The current implementation for torch assumes that model params are saved as 'model_state_dict'; if this is not true, `mode='torch'` fails, but `mode=None` simply returns the path to the checkpoint file so that the user can parse it differently. Not sure if there is a better way to handle that?

Reply:
Ok. We could also be opinionated about users having to save model params as a state_dict, but in the spirit of keeping things flexible and lightweight this probably makes more sense